packages/components/src/mobile/gradient/index.native.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * External dependencies
+ */
+import { View, Platform } from 'react-native';
+import RNLinearGradient from 'react-native-linear-gradient';
+/**
+ * WordPress dependencies
+ */
+import { colorsUtils } from '@wordpress/components';
+import { RadialGradient, Stop, SVG, Defs, Rect } from '@wordpress/primitives';
+import { useResizeObserver } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+
+function Gradient( {
+	gradientValue,
+	style,
+	angleCenter = { x: 0.5, y: 0.5 },
+	...otherProps
+} ) {
+	const [ resizeObserver, sizes ] = useResizeObserver();
+	const { width = 0, height = 0 } = sizes || {};
+	const { isGradient, getGradientType, gradients } = colorsUtils;
+
+	if ( ! gradientValue || ! isGradient( gradientValue ) ) {
+		return null;
+	}
+
+	const isLinearGradient =
+		getGradientType( gradientValue ) === gradients.linear;
+
+	const matchColorGroup = /(rgba|rgb|#)(.+?)[\%]/g;
+	const matchDeg = /(\d.+)deg/g;
+
+	const colorGroup = gradientValue
+		.match( matchColorGroup )
+		.map( ( color ) => color.split( ' ' ) );
+
+	const colors = colorGroup.map( ( color ) => color[ 0 ] );
+	const locations = colorGroup.map(
+		( location ) => Number( location[ 1 ].replace( '%', '' ) ) / 100
+	);
+
+	const angle =
+		isLinearGradient && Number( matchDeg.exec( gradientValue )[ 1 ] );
+
+	if ( isLinearGradient ) {
+		return (
+			<RNLinearGradient
+				colors={ colors }
+				useAngle={ true }
+				angle={ angle }
+				locations={ locations }
+				angleCenter={ angleCenter }
+				style={ style }
+				{ ...otherProps }
+			/>
+		);
+	}
+
+	return (
+		<View style={ [ style, styles.overflow ] }>
+			{ resizeObserver }
+			<SVG>
+				<Defs>
+					<RadialGradient
+						//eslint-disable-next-line no-restricted-syntax
+						id="radialGradient"
+						gradientUnits="userSpaceOnUse"
+						rx="70%"
+						ry="70%"
+						cy={ Platform.OS === 'android' ? width / 2 : '50%' }
+					>
+						{ colorGroup.map( ( group ) => {
+							return (
+								<Stop
+									offset={ group[ 1 ] }
+									stopColor={ group[ 0 ] }
+									stopOpacity="1"
+									key={ group[ 0 ] }
```

Comment:
Are we sure that the group [ 0 ] is unique? Can we have 2 the same colors in one gradient? If yes, we definitely should use a mix of offset and color, WDYT?

Reply:
It _shouldn't_ happen, but I'll do that just to be sure!

Reply:
Hmmm? What do you mean by `it shouldn't happen`? If you start your gradient by white and finish it by white then the first and last item will have the same key

Reply:
True that, I forgot that on web we can add custom color to gradient which can have the same value 👍, nice catch!